src/traders-controller.ts

```diff
@@ -80,6 +86,9 @@ export class TradersController {
           trader.base.loyaltyLevels.forEach((payloadLevel) => {
             payloadLevel.insurance_price_coef =
               insuranceTraderConfig.insurance_price_coef || 1;
+
+          insuranceConfig.returnChancePercent[traderId] =
+            insuranceTraderConfig.return_chance_percent || 0; // i keep this value (as well as the max_return times above) at 0 in my personal set up in case anybody wants to actually use the value 0, otherwise using 0 returns null
```

Comment:
Here I would add an `if` to check if `return_chance_percent` is not `undefined` or `null` before doing the mutation

```ts
if (insuranceTraderConfig.return_chance_percent !== undefined && insuranceTraderConfig.return_chance_percent !== null) {
  insuranceConfig.returnChancePercent[traderId] = insuranceTraderConfig.return_chance_percent
}
```

Otherwise, in a config that does not use this flag, the return chance will be set to zero for all traders and we don't want that here.

Reply:
makes sense - would you consider doing that for the max_return_time and min_return_time properties for the insurance as well? in case folks want instant returns

Reply:
Could worth it to make on every field (but it will need some testing to ensure it works as intended)